src/prefect/flows.py

```diff
@@ -246,8 +255,15 @@ def with_options(
         result_serializer: Optional[ResultSerializer] = NotSet,
         cache_result_in_memory: bool = None,
         log_prints: Optional[bool] = NotSet,
-        on_completion: Optional[List[Callable[[Flow, FlowRun, State], None]]] = None,
-        on_failure: Optional[List[Callable[[Flow, FlowRun, State], None]]] = None,
+        on_completion: Optional[
+            List[Callable[[schemas.core.Flow, schemas.core.FlowRun, State], None]]
+        ] = None,
+        on_failure: Optional[
+            List[Callable[[schemas.core.Flow, schemas.core.FlowRun, State], None]]
+        ] = None,
+        on_crashed: Optional[
+            List[Callable[[schemas.core.Flow, schemas.core.FlowRun, State], None]]
```

Comment:
The client should be working with `prefect.client.schemas.FlowRun` objects not the server-side ones

Reply:
Makes sense